sssd_test_framework/roles/ipa.py

```diff
@@ -1597,3 +1610,103 @@ def __get_info(self, info: str | NFSExport | IPAAutomountMap | None) -> str | No
+        :type name: str
+        """
+        super().__init__(role, name, command_group="pwpolicy")
+
+    def complexity(self, enable: bool) -> IPAPasswordPolicy:
+        """
+        Enable or disable password complexity.
+
+        :param enable: Enable or disable password complexity.
+        :type enable: bool
+        :return: IPAPasswordPolicy object.
+        :rtype: IPAPasswordPolicy
+        """
+        if enable:
+            attrs: CLIBuilderArgs = {
+                "dictcheck": (self.cli.option.VALUE, "True"),
+                "usercheck": (self.cli.option.VALUE, "True"),
+                "minlength": (self.cli.option.VALUE, 8),
+                "minclasses": (self.cli.option.VALUE, 5),
+                "priority": (self.cli.option.VALUE, 1),
+            }
+            self._add(attrs)
+        else:
+            _attrs: CLIBuilderArgs = {
+                "dictcheck": (self.cli.option.VALUE, "False"),
+                "usercheck": (self.cli.option.VALUE, "False"),
+                "minlength": (self.cli.option.VALUE, 0),
+                "minclasses": (self.cli.option.VALUE, 0),
+                "priority": (self.cli.option.VALUE, 1),
+            }
+            self._modify(_attrs)
+
+        return self
+
+    def lockout(self, duration: int, attempts: int) -> IPAPasswordPolicy:
+        """
+        Set lockout duration and login attempts.
+
+        :param duration: Duration of lockout in seconds.
+        :type duration: int
+        :param attempts: Number of login attempts.
+        :type attempts: int
+        :return: IPAPasswordPolicy object.
+        :rtype: IPAPasswordPolicy
+        """
+        attrs: CLIBuilderArgs = {
+            "lockouttime": (self.cli.option.VALUE, str(duration)),
+            "maxfail": (self.cli.option.VALUE, str(attempts)),
+        }
+        self._add(attrs)
+
+        return self
+
+    def age(self, minimum: int, maximum: int) -> IPAPasswordPolicy:
+        """
+        Set maximum and minimum password age.
+
+        :param minimum: Minimum password age in seconds, converted to days.
+        :type minimum: int
+        :param maximum: Maximum password age in seconds, converted to days.
+        :type maximum: int
+        :return: IPAPasswordPolicy object.
+        :rtype: IPAPasswordPolicy
+        """
+        attrs: CLIBuilderArgs = {
+            "minlife": (self.cli.option.VALUE, str(minimum)),
+            "maxlife": (self.cli.option.VALUE, str(maximum)),
+        }
+
+        self._add(attrs)
+
+        return self
+
+    def requirements(self, length: int) -> IPAPasswordPolicy:
+        """
+        Set password requirements, like length.
+
+        :param length: Required password character count.
+        :type length: int
+        :return: IPAPasswordPolicy object.
+        :rtype: IPAPasswordPolicy
+        """
+        attrs: CLIBuilderArgs = {
+            "minlength": (self.cli.option.VALUE, length),
+        }
+        self._add(attrs)
```

Comment:
Is it normal that attributes are added here (and in all the other methods) but the command receiving them is never executed?

Reply:
It is executed, in the test? I don't understand. 

```
    user = provider.user("user1").add(password="Secret123").password_change_at_logon
    provider.password.complexity(enable=True)
```

https://github.com/SSSD/sssd/pull/7728/files#diff-434faf79215d5c72ebbe450e17e1ce185261b8b2b5c96e932ba39311f04ab95dR118